### PR TITLE
validate attributes when creating SQS queues

### LINKED ISSUE #10812: bug: Cannot create an SQS queue with FifoQueue: false
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I can't create an SQS Queue using the AWS JavaScript SDK with FifoQueue: "false". Attempting to do so gives an error of `InvalidAttributeName: Unknown Attribute FifoQueue` and metadata saying that localstack responded with a 400.

### Expected Behavior

The queue is created successfully.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

docker-compose.yml

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```typescript
const sqs = new SQSClient({ endpoint: "http://localstack:4566", credentials: { accessKeyId: "foo", secretAccessKey: "bar", }, region: "us-east-1" });

await sqs.send(
    new CreateQueueCommand({
      QueueName: "test-queue",
      Attributes: {
        FifoQueue: "false",
      },
    }),
  );
```

### Environment

```markdown
- OS: Ubuntu 22.04
- LocalStack: 3.4.0
- @aws-sdk/client-sqs version: 3.370.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cloutierMat**
## Motivation

Looking into issue #10812, I realised that the attributes were not being validated at queue creation.

While the [documentation](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_CreateQueue.html#API_CreateQueue_RequestParameters) does mention `false` as an appropriate value for this attribute, reports are dating back 2017 that providing it results in `AttributeNameError`. 

## Changes

The method `validate_queue_attributes` was already enforcing this behaviour but it wasn't called when originally creating the queue.

A test for fifo queue attributes is also now fixed. :smile: 

fix #10812

### CODE REVIEW

File: tests/aws/services/cloudformation/resources/test_sqs.py [hunk 1]
```diff
@@ -33,8 +33,7 @@ def test_sqs_fifo_queue_generates_valid_name(deploy_cfn_template):
     assert ".fifo" in result.outputs["FooQueueName"]
 
 
-# FIXME: doesn't work on AWS. (known bug in cloudformation: https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/165)
-@markers.aws.unknown
+@markers.aws.validated
```
Comment (simonrw) [inline tests/aws/services/cloudformation/resources/test_sqs.py:36]: Woo! :tada:

**Review by simonrw — APPROVED** [review]
I will let the service owner(s) comment on the SQS changes, but the CFn changes look good to me. Just a non-critical suggestion for the future

File: localstack/services/sqs/models.py [hunk 2]
```diff
@@ -881,6 +873,23 @@ def _on_remove_message(self, message: SqsMessage):
                 # this may happen if the message no longer exists because it was removed earlier
                 pass
 
+    def validate_queue_attributes(self, attributes):
+        valid = [
+            k[1]
+            for k in inspect.getmembers(
+                QueueAttributeName, lambda x: isinstance(x, str) and not x.startswith("__")
```
Comment (baermat) [inline localstack/services/sqs/models.py:880]: purely curious, what's the reasoning behind the startswith clause, which attributes are weeded out here?

**Review by baermat — APPROVED** [review]
LGTM :+1: thanks for tackling this :)

File: localstack/services/sqs/models.py (hunk 2 above)
Comment (cloutierMat) [reply localstack/services/sqs/models.py:880]: `get_members` returns all of the members and dunder methods, after filtering for `str`, I believe it is `__name__` that was left.

**Pull request merged**